### PR TITLE
Utopia-511 - CSS standardization with UI/UX guidelines

### PR DESCRIPTION
**Pull request by antsand**
Have consistent naming convention for scss variables and button class using BEM standard.

This commit also fixes button, padding and footer issues.

# Description

This PR includes the following proposed change(s):

- {List all the changes, if possible add the jira ticket #}

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

**Comment by Bottle7** [comment]
I know this is super nit-picky but if we're following BEM naming—as referred to in the wiki—then naming should follow this naming template:

`block-name__elem-name--mod-name--mod-val`

File: src/frontend/src/sass/_variables.scss [hunk 1]
```diff
@@ -5,14 +5,35 @@
...
+
+/* Semantic colours */
+$error:#d8292f;
+$success:#2E8540;
+$warning: #EED202;
+$info: #1A5A96;
+$focus:#3B99FC;
+
+/* FORM Elements */
+$form-input-border-colour: #606060;
```
Comment (kushal-arora-fw) [inline src/frontend/src/sass/_variables.scss:39]: Missing newline

File: src/frontend/src/sass/_variables.scss [hunk 2]
```diff
@@ -5,14 +5,35 @@
...
+$bg-white: #ffffff;
+$bg-blue: #38598a;
+$bg-lightgray: #f2f2f2;
+
+/* Semantic colours */
+$error:#d8292f;
+$success:#2E8540;
+$warning: #EED202;
+$info: #1A5A96;
+$focus:#3B99FC;
```
Comment (kushal-arora-fw) [inline src/frontend/src/sass/_variables.scss:36]: nit: missing empty spaces between `:` & `#`

File: src/frontend/src/sass/_variables.scss [hunk 3]
```diff
@@ -5,14 +5,35 @@
...
 $font-family: 'BCSans', sans-serif;
-$bg-white: white;
-$bg-blue: #38598a;
 $line-height: 1.5;
-$primary-blue: #036;
-$link-blue: #1a5a96;
-$secondary-yellow: #fcba19;
-$ntrl-lightgray: #f2f2f2;
-$ntrl-darkgray: #313132;
+$secondary-yellow: #fcba19; /* !TODO: should be deleted once implementation changed to primary yellow */
```
Comment (kushal-arora-fw) [inline src/frontend/src/sass/_variables.scss:9]:
```suggestion
$secondary-yellow: #fcba19; /* !TODO: should be deleted once implementation changed to accent-colour */
```

**Comment by Bottle7** [comment]
I am also getting this behaviour on the PPQ Landing Page:

Screenshot 2022-11-22 at 12 35 35

**Comment by akroon3r** [comment]
I've put the DO NOT MERGE tag on this PR. 

Please make sure there's a ticket in our JIRA backlog associated with this work, and if there is, make sure that the title of this PR reflects that via the name.

**Review by kushal-arora-fw — APPROVED** [review]
Overall, looks good! Added some minor comments

**Comment by kushal-arora-fw** [comment]
> I am also getting this behaviour on the PPQ Landing Page:
> 
> Screenshot 2022-11-22 at 12 35 35

This isn't the case for me. I don't see any blank white space at the bottom.

File: src/frontend/src/components/common/Modal/_modal.scss [hunk 4]
```diff
@@ -41,6 +41,16 @@
...
     border-radius: 5px;
   }
+
+  .modalbtn__container {
+    display:flex;
+    flex-direction: row;
+    justify-content:center;
+    button {
+      margin-right:4px;
+      margin-left:4px;
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/common/Modal/_modal.scss:51]: spacing inconsistencies between `:` and the value

File: src/frontend/src/sass/_buttons.scss [hunk 5]
```diff
@@ -48,14 +49,14 @@ a:hover {
...
+.bcgovbtn:focus,
+.bcgovbtn:focus-visible {
   outline: 4px auto -webkit-focus-ring-color;
 }
 
-.bcgovbtn--primary {
+.bcgovbtn__primary {
   background-color: $primary-colour;
-  color: $text-light;
+  --text-color: #{$text-light};
```
Comment (kushal-arora-fw) [inline src/frontend/src/sass/_buttons.scss:59]: wondering what adding `--` in front of `text-color` does?

Comment (antsand) [reply src/frontend/src/sass/_buttons.scss:59]: Oh! it is a CSS variable. We are trying to use both CSS custom properties and sass variables. It makes our css cleaner.
https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties

**Review by kushal-arora-fw — APPROVED** [review]

**Review by Bottle7 — APPROVED** [review]

**Pull request merged**